usr/src/uts/common/fs/zfs/dmu.c

```diff
@@ -651,22 +660,60 @@ get_next_chunk(dnode_t *dn, uint64_t *start, uint64_t minimum)
 	return (0);
 }
 
+/*
+ * If this objset is of type OST_ZFS return true if vfs's unmounted flag is set,
+ * otherwise return false.
+ * Used below in dmu_free_long_range_impl() to enable abort on ZFS unmount
+ */
+/*ARGSUSED*/
+static boolean_t
+dmu_objset_zfs_unmounting(objset_t *os)
+{
+#ifdef _KERNEL
+	if (dmu_objset_type(os) == DMU_OST_ZFS)
+		return (zfs_get_vfs_flag_unmounted(os));
+#endif
+	return (B_FALSE);
+}
+
 static int
 dmu_free_long_range_impl(objset_t *os, dnode_t *dn, uint64_t offset,
     uint64_t length)
 {
 	uint64_t object_size = (dn->dn_maxblkid + 1) * dn->dn_datablksz;
 	int err;
+	uint8_t dirty_frees_threshold;
+	dsl_pool_t *dp = dmu_objset_pool(os);
+
 
 	if (offset >= object_size)
 		return (0);
 
+	if (zfs_per_txg_dirty_frees_percent <= 100)
+		dirty_frees_threshold = zfs_per_txg_dirty_frees_percent;
+	else
+		dirty_frees_threshold = MAX(1, zfs_delay_min_dirty_percent / 4);
```

Comment:
This is to provide a "reasonable" default if zfs_per_txg_dirty_frees_percent has been set to an invalid percent (e.g. 12345)?